packages/node_modules/@ciscospark/react-component-activity-share-files/src/index.js

```diff
@@ -49,24 +49,14 @@ export function ActivityShareFiles(props) {
           objectUrl = thumbnail.get('objectUrl');
         }
       }
-
-      return (
-        <ShareThumbnail
-          file={file}
-          isFetching={isFetching}
-          isPending={isPending}
-          key={file.url}
-          objectUrl={objectUrl}
-          onDownloadClick={onDownloadClick}
-        />
-      );
     }
     return (
-      <ShareFile
-        file={file}
-        isPending={isPending}
-        key={file.url}
-        onDownloadClick={onDownloadClick}
+      <FileShareDisplay
+        {
+          ...{
+            file, isFetching, isPending, key, objectUrl, onDownloadClick
+          }
```

Comment:
This style feels gross to me, but I don't know why. Could constructing a fileShareProps object be better?

Reply:
I was attempting to implement [this](https://medium.com/@tevthuku/less-code-defining-your-react-props-eb093c7d3768) but I'm pretty sure it feels gross to me too, lol.